core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1034,13 +1047,17 @@ class WorkspaceService(
               )
             }
 
-            (clonedEntityCount, clonedAttrCount) <- dataAccess.entityQuery.copyEntitiesToNewWorkspace(
-              sourceWorkspaceContext.workspaceIdAsUUID,
-              destWorkspaceContext.workspaceIdAsUUID
+            entityProvider <- DBIO.from(
+              entityServiceConstructor(ctx).getProviderWithTracing(sourceWorkspaceContext, ctx)
             )
-
-            _ = clonedWorkspaceEntityHistogram += clonedEntityCount
-            _ = clonedWorkspaceAttributeHistogram += clonedAttrCount
+            _ <- traceDBIOWithParent("clone entities", ctx) { s =>
+              entityProvider
+                .clone(sourceWorkspaceContext, destWorkspaceContext, s)
+                .map { case (clonedEntityCount, clonedAttrCount) =>
+                  clonedWorkspaceEntityHistogram += clonedEntityCount
+                  clonedWorkspaceAttributeHistogram += clonedAttrCount
+                }
+            }
```

Comment:
Since you have access to `EntityService` via `entityServiceConstructor(ctx)` now, how about moving all of this logic into a method on `EntityService`?

Try a "find usages" in IntelliJ on `getProviderWithTracing` - it's called 17 times from within EntityService and then once from here. Moving this logic into EntityService would be more consistent - it would also allow you to return `getProviderWithTracing` to private.